src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -90,7 +90,14 @@ internal static bool IsEnabled()
                         do
                         {
                             KeyValuePair<string, string> item = e.Current;
-                            baggage.Add(new NameValueHeaderValue(item.Key, item.Value).ToString());
+                            try
+                            {
+                                baggage.Add(new NameValueHeaderValue(item.Key, item.Value).ToString());
+                            }
+                            catch (FormatException)
```

Comment:
Why ignore these malformed tokens?  Is it possible to log them in some way?  Does DiagnosticsListener only care about well-formed tokens.

It seem like our diagnostic tools should be able to log malformed tokens.  Isn't that the point of diagnostics? :)

Reply:
@davidsh
These tokens are the arbitrary context application adds into the Activity.Baggage.

Baggage is propagated to the downstream services over the wire.
It would be reasonable to validate them when they are added into the baggage, but the user may have got them from the upstream service which is hot HTTP and token were valid at that time. Baggage is added by the framework/library (e.g. grpc or queue message handler) and this could be out the user's of control.
So Activity.Baggage does not know the protocol over which tokens will be propagated and cannot control its format. 

Throwing on invalid token does not make sense as baggage carries diagnostics information and breaking business logic because of that is not reasonable.

Users will get the first chance exception that should allow them to investigate missing baggages - this is the concept Activity implements.

Reply:
Thanks for the added detail.  It would be useful to add a summary of this as a comment inside the empty `catch (FormatException)` block.

Reply:
sure, will do